### PR TITLE
Expose WithRetryPolicy in workflow package

### PR DESCRIPTION
**Pull request by sagikazarmark**
Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

### CODE REVIEW

File: workflow/activity_options.go [hunk 1]
```diff
@@ -71,3 +74,8 @@ func WithHeartbeatTimeout(ctx Context, d time.Duration) Context {
 func WithWaitForCancellation(ctx Context, wait bool) Context {
 	return internal.WithWaitForCancellation(ctx, wait)
 }
+
+// WithRetryPolicy adds wait for the cacellation to the copy of the context.
```
Comment (sevein) [inline workflow/activity_options.go:78]:
```suggestion
// WithRetryPolicy adds wait for the cancellation to the copy of the context.
```

Comment (sagikazarmark) [reply workflow/activity_options.go:78]: You fixed a typo in a copy paste error. 😆

**Comment by mfateev** [comment]
RetryPolicy is already exposed: https://github.com/uber-go/cadence-client/blob/master/cadence.go

**Comment by sagikazarmark** [comment]
The type itself is, but that's a different package and it would lead to an import cycle.

This PR is about the `WithRetryPolicy` context mutator.

**Comment by sagikazarmark** [comment]
Thanks!

**Pull request merged**